pkg/apis/build/v1alpha1/build_types.go

```diff
@@ -159,15 +165,15 @@ type Image struct {
 type BuildStatus struct {
 	// The Register status of the Build
 	// +optional
-	Registered corev1.ConditionStatus `json:"registered,omitempty"`
+	Registered *corev1.ConditionStatus `json:"registered,omitempty"`
```

Comment:
I would not make changes like those. I'd say an optional string field where we do not need to distinguish `nil` and `""`, can be a non-pointer with omitempty. But we should make this a guidance.

But seeking for other opinions on this @gabemontero @adambkaplan @HeavyWombat

Reply:
I agree that we should leave this as a non-pointer for two reasons:

1. Empty string for `ConditionStatus` is always meaningless in our code - there are explict values for True/False/Unknown
2. Status should only be updated by our controllers - there is no user intent to decipher.

Reply:
>we ask that pointers always be used with optional fields that do not have a built-in nil value

Better to keep pointers for optional fields,  [api-conventions](https://github.com/kubernetes/community/blob/master/contributors/devel/sig-architecture/api-conventions.md#optional-vs-required) also suggest the same.

Also, we had discussed this in a [community meeting](https://github.com/shipwright-io/community/issues/42#issuecomment-975581815) and the decision was to go with pointers for the optional fields.